.github/workflows/modular-arch-quality-gates.yml

```
@@ -0,0 +1,423 @@
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Detect changed modules
+        id: check-changes
+        run: |
+          # Use GitHub API to get the actual files changed in this PR
+          # This avoids issues with merge base and branch history
+          PR_NUMBER="${{ github.event.pull_request.number }}"
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)